cdap-app-fabric/src/main/java/io/cdap/cdap/internal/credential/handler/CredentialProviderHttpHandler.java

```diff
@@ -396,6 +421,10 @@ private void ensureNamespaceExists(String namespace)
     Boolean namespaceFound;
     try {
       namespaceId = new NamespaceId(namespace);
+      if (NamespaceId.SYSTEM.equals(namespaceId)) {
```

Comment:
Should we also put this logic in DefaultNamespaceAdmin? Additionally, should we add all reserved namespaces here?

Reply:
I do not see any obvious reason of extending to all reserved namespaces here since, `default` namespace is present in the namespaces table created by bootstrap class.

Since we do not allow creation of namespace with name `cdap`, there is no reason of adding that as well.

I added only `SYSTEM` namespace, since we will be adding a profile in `SYSTEM` namespace.

WDYT?

Reply:
Ah I see, so we're doing this specifically to prevent the creation of identities in other reserved namespace that might not exist. In that case, LGTM.

Reply:
ack, done.